app/src/main/java/fr/free/nrw/commons/upload/LanguagesAdapter.kt

```diff
@@ -86,7 +89,9 @@ class LanguagesAdapter constructor(
     }
 
     fun getIndexOfUserDefaultLocale(context: Context): Int {
-        return language.codes.indexOf(context.locale!!.language)
+
+        val userLanguageCode = context.locale?.language ?: return DEFAULT_INDEX
```

Comment:
Would you mind linking to the bug and explaining what was the problem?
That will prevent future contributors from "simplifying" that function. :-)

Reply:
Sure, I will add that soon.

Reply:
Explanation and comment have been added.